doc/proxy-injection.md

```diff
@@ -1,94 +1,66 @@
+existed whereby admisson controllers could be maintained
+out-of-tree. There is no platform support for this yet, but a proposal
+has been created to add such a feature
+(see [Proposal: Extensible Admission Control](https://github.com/kubernetes/community/pull/132/)).
+
+Long term istio automatic proxy injection is being tracked
+by [Kubernetes Admission Controller for proxy injection](https://github.com/istio/manager/issues/57).
+
 ## Manual injection
 
-Add the following annotation to the deployment's template metadata.
+A short term workaround for the lack of a proper istio admision
+controller is client-side injection. Use `istioctl kube-inject` to add the
+necessary configurations to a kubernetes resource files.
 
-```
-annotations:
-  pod.beta.kubernetes.io/init-containers: >
-    [{
-      "name": "init-proxy",
-      "image": "docker.io/istio/init:latest",
-      "imagePullPolicy": "Always",
-      "securityContext": { "capabilities" : { "add" : ["NET_ADMIN"] } }
-    }]
-```
+    istioctl kube-inject -f deployment.yaml -o deployment-with-istio.yaml
 
-Add the following container to the deployment's template spec. The UID '1337' should match ISTIO_PROXY_UID as specified by [proxy-redirection-configuration.md](proxy-redirection-configuration.md )
+Or update the resource on the fly before applying.
 
-```
-- name: proxy
-  image: docker.io/istio/runtime:latest
-  imagePullPolicy: Always
-  securityContext:
-    runAsUser: 1337
-  args:
-    - proxy
-    - -s
-    - manager:8080
-```
+    istioctl kube-inject -f depoyment.yaml | kubectl appy -f -
 
-A sample deployment with manually injected init and proxy containers might look like this.
+Or update an existing deployment.
 
-```
-apiVersion: v1
-kind: Service
-metadata:
-  name: example
-  labels:
-    app: example
-spec:
-  type: NodePort
-  ports:
-  - port: 80
-    targetPort: 8080
-    name: http
-  selector:
-    app: example
----
-apiVersion: extensions/v1beta1
-kind: Deployment
-metadata:
-  name: example
-spec:
-  replicas: 1
-  template:
-    metadata:
-      labels:
-        app: example
-      annotations:
-        pod.beta.kubernetes.io/init-containers: >
-          [{
-            "name": "proxy-init",
-            "image": "docker.io/istio/init:latest",
-            "imagePullPolicy": "Always",
-            "securityContext": { "capabilities" : { "add" : ["NET_ADMIN"] } }
-          }]
-    spec:
-      containers:
-      - name: app
-        image: docker.io/istio/app:latest
-	imagePullPolicy: Always
-        args:
-          - --port
-          - "8080"
-        ports:
-        - containerPort: 8080
-      - name: proxy
-        image: docker.io/istio/runtime:latest
-        imagePullPolicy: Always
-        securityContext:
-          runAsUser: 1337
-        args:
-          - proxy
-          - -s
-          - manager:8080
----
-```
+    kubectl get deployment -o yaml | istioctl kube-inject -f - | kubectl apply -f -
```

Comment:
How do you mean? `kubectl edit` is interactive so we'd need to modify KUBE_EDITOR to use `istioctl kube-inject`. On the other hand, `kubectl edit` in interactive mode does show the resource with injected proxy.

```
$ KUBE_EDITOR="grep 'alpha.istio.io/sidecar: injected'" kubectl edit -f in.yaml
        alpha.istio.io/sidecar: injected
```